### ISSUE TITLE
Yotta build causes exception "WindowsError: [Error 183] Cannot create a file when that file already exists"

### ISSUE THREAD

**Issue by mazimkhan** [ARMmbed/yotta#707]
Yotta build causes below exception. 

Seen after upgrading cmake to version 3.4, ninja to version 1.6 on request in #701 

```
09:57:04 Started by upstream project "bt" build number 5
09:57:04 originally caused by:
09:57:04  Started by upstream project "mbed-drivers-nightly-build-and-test2/compiler=armcc,target=frdm-k64f-armcc" build number 8
09:57:04  originally caused by:
09:57:04   Started by upstream project "mbed-drivers-nightly-build-and-test2" build number 8
09:57:04   originally caused by:
09:57:04    Started by user mazimkhan
09:57:04 [EnvInject] - Loading node environment variables.
09:57:04 Building remotely on cam_int_win_hw_3_builds (gcc windows armcc) in workspace c:\jslv3_1\ws\yb@3
09:57:04 [WS-CLEANUP] Deleting project workspace...
09:57:11 [WS-CLEANUP] Done
09:57:11 Cloning the remote Git repository
09:57:11 Cloning repository git@github.com:ARMmbed/mbed-drivers.git
09:57:11  > C:\Program Files\Git\cmd\git.exe init c:\jslv3_1\ws\yb@3 # timeout=10
09:57:11 Fetching upstream changes from git@github.com:ARMmbed/mbed-drivers.git
09:57:11  > C:\Program Files\Git\cmd\git.exe --version # timeout=10
09:57:11 using GIT_SSH to set credentials Jenkins master ssh keys
09:57:11  > C:\Program Files\Git\cmd\git.exe -c core.askpass=true fetch --tags --progress git@github.com:ARMmbed/mbed-drivers.git +refs/heads/*:refs/remotes/origin/*
09:59:30  > C:\Program Files\Git\cmd\git.exe config remote.origin.url git@github.com:ARMmbed/mbed-drivers.git # timeout=10
09:59:30  > C:\Program Files\Git\cmd\git.exe config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
09:59:30  > C:\Program Files\Git\cmd\git.exe config remote.origin.url git@github.com:ARMmbed/mbed-drivers.git # timeout=10
09:59:30 Fetching upstream changes from git@github.com:ARMmbed/mbed-drivers.git
09:59:30 using GIT_SSH to set credentials Jenkins master ssh keys
09:59:30  > C:\Program Files\Git\cmd\git.exe -c core.askpass=true fetch --tags --progress git@github.com:ARMmbed/mbed-drivers.git +refs/heads/*:refs/remotes/origin/*
09:59:31  > C:\Program Files\Git\cmd\git.exe rev-parse "refs/remotes/origin/master^{commit}" # timeout=10
09:59:31  > C:\Program Files\Git\cmd\git.exe rev-parse "refs/remotes/origin/origin/master^{commit}" # timeout=10
09:59:31 Checking out Revision bfdff4582c170f72fee86ed2ab5fbe3b602a2c0b (refs/remotes/origin/master)
09:59:31  > C:\Program Files\Git\cmd\git.exe config core.sparsecheckout # timeout=10
09:59:31  > C:\Program Files\Git\cmd\git.exe checkout -f bfdff4582c170f72fee86ed2ab5fbe3b602a2c0b
09:59:32  > C:\Program Files\Git\cmd\git.exe rev-list bfdff4582c170f72fee86ed2ab5fbe3b602a2c0b # timeout=10
09:59:32 Cloning the remote Git repository
09:59:32 Cloning repository git@github.com:ARMmbed/update-tools.git
09:59:32  > C:\Program Files\Git\cmd\git.exe init c:\jslv3_1\ws\yb@3\update-tools # timeout=10
09:59:33 Fetching upstream changes from git@github.com:ARMmbed/update-tools.git
09:59:33  > C:\Program Files\Git\cmd\git.exe --version # timeout=10
09:59:33 using GIT_SSH to set credentials Jenkins master ssh keys
09:59:33  > C:\Program Files\Git\cmd\git.exe -c core.askpass=true fetch --tags --progress git@github.com:ARMmbed/update-tools.git +refs/heads/*:refs/remotes/origin/*
09:59:38  > C:\Program Files\Git\cmd\git.exe config remote.origin.url git@github.com:ARMmbed/update-tools.git # timeout=10
09:59:39  > C:\Program Files\Git\cmd\git.exe config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
09:59:39  > C:\Program Files\Git\cmd\git.exe config remote.origin.url git@github.com:ARMmbed/update-tools.git # timeout=10
09:59:39 Fetching upstream changes from git@github.com:ARMmbed/update-tools.git
09:59:39 using GIT_SSH to set credentials Jenkins master ssh keys
09:59:39  > C:\Program Files\Git\cmd\git.exe -c core.askpass=true fetch --tags --progress git@github.com:ARMmbed/update-tools.git +refs/heads/*:refs/remotes/origin/*
09:59:40  > C:\Program Files\Git\cmd\git.exe rev-parse "refs/remotes/origin/master^{commit}" # timeout=10
09:59:40  > C:\Program Files\Git\cmd\git.exe rev-parse "refs/remotes/origin/origin/master^{commit}" # timeout=10
09:59:40 Checking out Revision 352babe8f22f89a18918653fcf099420e54e4205 (refs/remotes/origin/master)
09:59:40  > C:\Program Files\Git\cmd\git.exe config core.sparsecheckout # timeout=10
09:59:40  > C:\Program Files\Git\cmd\git.exe checkout -f 352babe8f22f89a18918653fcf099420e54e4205
09:59:40  > C:\Program Files\Git\cmd\git.exe rev-list 352babe8f22f89a18918653fcf099420e54e4205 # timeout=10
09:59:40 Next nodes: []
09:59:41 [yb@3] $ sh -xe C:\Windows\TEMP\hudson3474165582156421340.sh
09:59:41 + rm -rf venv
09:59:41 + virtualenv venv
09:59:43 New python executable in c:\jslv3_1\ws\yb@3\venv\Scripts\python.exe
10:00:21 Installing setuptools, pip, wheel...done.
10:00:21 + source venv/Scripts/activate
10:00:21 ++ deactivate nondestructive
10:00:21 ++ unset -f pydoc
10:00:21 ++ '[' -z '' ']'
10:00:21 ++ '[' -z '' ']'
10:00:21 ++ '[' -n /usr/bin/sh ']'
10:00:21 ++ hash -r
10:00:21 ++ '[' -z '' ']'
10:00:21 ++ unset VIRTUAL_ENV
10:00:21 ++ '[' '!' nondestructive = nondestructive ']'
10:00:21 +++ '[' msys == cygwin ']'
10:00:21 +++ echo /c/jslv3_1/ws/yb@3/venv
10:00:21 ++ VIRTUAL_ENV=/c/jslv3_1/ws/yb@3/venv
10:00:21 ++ export VIRTUAL_ENV
10:00:21 ++ _OLD_VIRTUAL_PATH='/c/Program Files (x86)/Windows Resource Kits/Tools:/c/ProgramData/Oracle/Java/javapath:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Python27:/c/Python27/Scripts:/cmd:/mingw64/bin:/usr/bin:/c/Users/IOTAdmin/Downloads/PSTools:/c/Program Files (x86)/CMake/bin:/c/ninja-win:/c/Program Files (x86)/GNU Tools ARM Embedded/4.9 2015q3/bin:/c/Program Files (x86)/ARM_Compiler_5/bin:/c/Program Files/Microsoft SQL Server/110/Tools/Binn:/c/Program Files (x86)/Microsoft SDKs/TypeScript/1.0:/c/Program Files/Microsoft SQL Server/120/Tools/Binn'
10:00:21 ++ PATH='/c/jslv3_1/ws/yb@3/venv/Scripts:/c/Program Files (x86)/Windows Resource Kits/Tools:/c/ProgramData/Oracle/Java/javapath:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Python27:/c/Python27/Scripts:/cmd:/mingw64/bin:/usr/bin:/c/Users/IOTAdmin/Downloads/PSTools:/c/Program Files (x86)/CMake/bin:/c/ninja-win:/c/Program Files (x86)/GNU Tools ARM Embedded/4.9 2015q3/bin:/c/Program Files (x86)/ARM_Compiler_5/bin:/c/Program Files/Microsoft SQL Server/110/Tools/Binn:/c/Program Files (x86)/Microsoft SDKs/TypeScript/1.0:/c/Program Files/Microsoft SQL Server/120/Tools/Binn'
10:00:22 ++ export PATH
10:00:22 ++ '[' -z '' ']'
10:00:22 ++ '[' -z '' ']'
10:00:22 ++ _OLD_VIRTUAL_PS1=
10:00:22 ++ '[' x '!=' x ']'
10:00:22 +++ basename /c/jslv3_1/ws/yb@3/venv
10:00:22 ++ PS1='(venv) '
10:00:22 ++ export PS1
10:00:22 ++ alias pydoc
10:00:22 ++ '[' -n /usr/bin/sh ']'
10:00:22 ++ hash -r
10:00:22 + test -e update-tools/build_requirements.txt
10:00:22 + pip install -r update-tools/build_requirements.txt
10:00:22 Collecting argcomplete==1.0.0 (from -r update-tools/build_requirements.txt (line 1))
10:00:23   Using cached argcomplete-1.0.0-py2.py3-none-any.whl
10:00:23 Collecting cffi==1.5.0 (from -r update-tools/build_requirements.txt (line 2))
10:00:23   Using cached cffi-1.5.0-cp27-none-win32.whl
10:00:23 Collecting colorama==0.3.6 (from -r update-tools/build_requirements.txt (line 3))
10:00:23   Using cached colorama-0.3.6-py2.py3-none-any.whl
10:00:23 Collecting cryptography==1.2.2 (from -r update-tools/build_requirements.txt (line 4))
10:00:24   Using cached cryptography-1.2.2-cp27-none-win32.whl
10:00:24 Collecting enum34==1.1.2 (from -r update-tools/build_requirements.txt (line 5))
10:00:24 Collecting functools32==3.2.3.post2 (from -r update-tools/build_requirements.txt (line 6))
10:00:24 Collecting hgapi==1.7.2 (from -r update-tools/build_requirements.txt (line 7))
10:00:24 Collecting idna==2.0 (from -r update-tools/build_requirements.txt (line 8))
10:00:24   Using cached idna-2.0-py2.py3-none-any.whl
10:00:24 Collecting intelhex==2.0 (from -r update-tools/build_requirements.txt (line 9))
10:00:24   Using cached intelhex-2.0-py2.py3-none-any.whl
10:00:24 Collecting ipaddress==1.0.16 (from -r update-tools/build_requirements.txt (line 10))
10:00:24   Using cached ipaddress-1.0.16-py27-none-any.whl
10:00:24 Collecting Jinja2==2.8 (from -r update-tools/build_requirements.txt (line 11))
10:00:24   Using cached Jinja2-2.8-py2.py3-none-any.whl
10:00:25 Collecting jsonpointer==1.10 (from -r update-tools/build_requirements.txt (line 12))
10:00:25   Using cached jsonpointer-1.10-py2-none-any.whl
10:00:25 Collecting jsonschema==2.5.1 (from -r update-tools/build_requirements.txt (line 13))
10:00:25   Using cached jsonschema-2.5.1-py2.py3-none-any.whl
10:00:25 Collecting MarkupSafe==0.23 (from -r update-tools/build_requirements.txt (line 14))
10:00:25 Collecting mbed-test-wrapper==0.0.2 (from -r update-tools/build_requirements.txt (line 15))
10:00:25 Collecting ndg-httpsclient==0.4.0 (from -r update-tools/build_requirements.txt (line 16))
10:00:25 Collecting ntfsutils==0.1.3 (from -r update-tools/build_requirements.txt (line 17))
10:00:25 Collecting pathlib==1.0.1 (from -r update-tools/build_requirements.txt (line 18))
10:00:26 Collecting project-generator==0.8.7 (from -r update-tools/build_requirements.txt (line 19))
10:00:26 Collecting project-generator-definitions==0.2.4 (from -r update-tools/build_requirements.txt (line 20))
10:00:27 Collecting pyasn1==0.1.9 (from -r update-tools/build_requirements.txt (line 21))
10:00:27   Using cached pyasn1-0.1.9-py2.py3-none-any.whl
10:00:27 Collecting pycparser==2.14 (from -r update-tools/build_requirements.txt (line 22))
10:00:27 Collecting pyelftools==0.23 (from -r update-tools/build_requirements.txt (line 23))
10:00:28 Collecting PyGithub==1.26.0 (from -r update-tools/build_requirements.txt (line 24))
10:00:31 Collecting PyJWT==1.4.0 (from -r update-tools/build_requirements.txt (line 25))
10:00:31   Using cached PyJWT-1.4.0-py2.py3-none-any.whl
10:00:31 Collecting pyOCD==0.6.1 (from -r update-tools/build_requirements.txt (line 26))
10:00:32 Collecting pyOpenSSL==0.15.1 (from -r update-tools/build_requirements.txt (line 27))
10:00:32   Using cached pyOpenSSL-0.15.1-py2.py3-none-any.whl
10:00:32 Collecting pywinusb==0.4.0 (from -r update-tools/build_requirements.txt (line 28))
10:00:33 Collecting PyYAML==3.11 (from -r update-tools/build_requirements.txt (line 29))
10:00:33 Collecting requests==2.9.1 (from -r update-tools/build_requirements.txt (line 30))
10:00:33   Using cached requests-2.9.1-py2.py3-none-any.whl
10:00:33 Collecting semantic-version==2.4.2 (from -r update-tools/build_requirements.txt (line 31))
10:00:33 Collecting six==1.10.0 (from -r update-tools/build_requirements.txt (line 32))
10:00:33   Using cached six-1.10.0-py2.py3-none-any.whl
10:00:33 Collecting valinor==0.0.14 (from -r update-tools/build_requirements.txt (line 33))
10:00:34 Requirement already satisfied (use --upgrade to upgrade): wheel==0.26.0 in .\venv\lib\site-packages (from -r update-tools/build_requirements.txt (line 34))
10:00:34 Collecting xmltodict==0.9.2 (from -r update-tools/build_requirements.txt (line 35))
10:00:34 Collecting yotta==0.13.1 (from -r update-tools/build_requirements.txt (line 36))
10:00:34 Requirement already satisfied (use --upgrade to upgrade): setuptools>=1.0 in .\venv\lib\site-packages (from cryptography==1.2.2->-r update-tools/build_requirements.txt (line 4))
10:00:34 Collecting argparse (from project-generator==0.8.7->-r update-tools/build_requirements.txt (line 19))
10:00:34   Using cached argparse-1.4.0-py2.py3-none-any.whl
10:00:34 Installing collected packages: argcomplete, pycparser, cffi, colorama, enum34, ipaddress, pyasn1, six, idna, cryptography, functools32, hgapi, intelhex, MarkupSafe, Jinja2, jsonpointer, jsonschema, mbed-test-wrapper, pyOpenSSL, ndg-httpsclient, ntfsutils, pathlib, argparse, xmltodict, PyYAML, project-generator-definitions, project-generator, pyelftools, PyGithub, PyJWT, pywinusb, pyOCD, requests, semantic-version, valinor, yotta
10:02:57 Successfully installed Jinja2-2.8 MarkupSafe-0.23 PyGithub-1.26.0 PyJWT-1.4.0 PyYAML-3.11 argcomplete-1.0.0 argparse-1.4.0 cffi-1.5.0 colorama-0.3.6 cryptography-1.2.2 enum34-1.1.2 functools32-3.2.3.post2 hgapi-1.7.2 idna-2.0 intelhex-2.0 ipaddress-1.0.16 jsonpointer-1.10 jsonschema-2.5.1 mbed-test-wrapper-0.0.2 ndg-httpsclient-0.4.0 ntfsutils-0.1.3 pathlib-1.0.1 project-generator-0.8.7 project-generator-definitions-0.2.4 pyOCD-0.6.1 pyOpenSSL-0.15.1 pyasn1-0.1.9 pycparser-2.14 pyelftools-0.23 pywinusb-0.4.0 requests-2.9.1 semantic-version-2.4.2 six-1.10.0 valinor-0.0.14 xmltodict-0.9.2 yotta-0.13.1
10:03:22 + yotta target frdm-k64f-armcc
10:03:23 info: get versions for frdm-k64f-armcc
10:03:24 info: download frdm-k64f-armcc@2.0.0 from the public module registry
10:03:25 info: get versions for kinetis-k64-armcc
10:03:25 info: download kinetis-k64-armcc@2.0.0 from the public module registry
10:03:26 info: get versions for mbed-armcc
10:03:26 info: download mbed-armcc@1.0.1 from the public module registry
10:03:27 + yotta build
10:03:28 info: get versions for mbed-hal
10:03:29 info: download mbed-hal@1.2.2 from the public module registry
10:03:44 info: get versions for cmsis-core
10:03:45 info: download cmsis-core@1.1.2 from the public module registry
10:03:45 info: get versions for ualloc
10:03:46 info: download ualloc@1.0.3 from the public module registry
10:03:46 info: get versions for minar
10:03:47 info: download minar@1.0.4 from the public module registry
10:03:56 info: get versions for core-util
10:03:57 info: download core-util@1.2.0 from the public module registry
10:03:57 info: get versions for compiler-polyfill
10:03:57 info: download compiler-polyfill@1.2.1 from the public module registry
10:03:57 info: get versions for mbed-hal-freescale
10:03:58 info: download mbed-hal-freescale@1.0.0 from the public module registry
10:04:05 info: get versions for mbed-hal-ksdk-mcu
10:04:06 info: download mbed-hal-ksdk-mcu@1.0.8 from the public module registry
10:04:06 info: get versions for uvisor-lib
10:04:07 info: download uvisor-lib@1.0.12 from the public module registry
10:04:07 info: get versions for mbed-hal-k64f
10:04:08 info: download mbed-hal-k64f@1.1.0 from the public module registry
10:04:09 Fatal Exception, yotta=0.13.1
10:04:09 Traceback (most recent call last):
10:04:09   File "C:\Python27\Lib\runpy.py", line 162, in _run_module_as_main
10:04:09     "__main__", fname, loader, pkg_name)
10:04:09   File "C:\Python27\Lib\runpy.py", line 72, in _run_code
10:04:09     exec code in run_globals
10:04:09   File "C:\jslv3_1\ws\yb@3\venv\Scripts\yotta.exe\__main__.py", line 9, in <module>
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\main.py", line 206, in main
10:04:09     status = args.command(args, following_args)
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\build.py", line 46, in execCommand
10:04:09     status = installAndBuild(args, following_args)
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\build.py", line 98, in installAndBuild
10:04:09     install_status = install.execCommand(args, [])
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\install.py", line 62, in execCommand
10:04:09     return installDeps(args, c)
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\install.py", line 119, in installDeps
10:04:09     test = {'own':'toplevel', 'all':True, 'none':False}[args.install_test_deps]
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\component.py", line 651, in satisfyDependenciesRecursive
10:04:09     test = test
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\component.py", line 436, in __getDependenciesRecursiveWithProvider
10:04:09     _processed = _processed
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\component.py", line 436, in __getDependenciesRecursiveWithProvider
10:04:09     _processed = _processed
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\component.py", line 436, in __getDependenciesRecursiveWithProvider
10:04:09     _processed = _processed
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\component.py", line 415, in __getDependenciesRecursiveWithProvider
10:04:09     test = test
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\component.py", line 315, in __getDependenciesWithProvider
10:04:09     satisfyDep, specs
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\component.py", line 291, in satisfyDep
10:04:09     self
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\component.py", line 636, in provider
10:04:09     inherit_shrinkwrap = dep_of.getShrinkwrap()
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\access.py", line 307, in satisfyVersionByInstalling
10:04:09     name, version_required, install_into, v, type=type, inherit_shrinkwrap = inherit_shrinkwrap
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\access.py", line 318, in _satisfyVersionByInstallingVersion
10:04:09     version.unpackInto(working_directory)
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\registry_access.py", line 447, in unpackInto
10:04:09     _getTarball(self.url, directory, self.sha256)
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\registry_access.py", line 100, in wrapped
10:04:09     return fn(*args, **kwargs)
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\registry_access.py", line 185, in wrapped
10:04:09     return fn(*args, **kwargs)
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\registry_access.py", line 168, in wrapped
10:04:09     return fn(*args, **kwargs)
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\registry_access.py", line 149, in wrapped
10:04:09     return fn(*args, **kwargs)
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\registry_access.py", line 296, in _getTarball
10:04:09     origin_info = {'url':url}
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\access_common.py", line 124, in wrapped
10:04:09     r = fn(*args, **kwargs)
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\access_common.py", line 288, in unpackTarballStream
10:04:09     new_cache_key = downloadToCache(stream, hash, cache_key, origin_info)
10:04:09   File "c:\jslv3_1\ws\yb@3\venv\lib\site-packages\yotta\lib\access_common.py", line 261, in downloadToCache
10:04:09     os.rename(download_fname, cache_as)
10:04:09 WindowsError: [Error 183] Cannot create a file when that file already exists
10:04:10 Build step 'Execute shell' marked build as failure
10:04:10 Archiving artifacts
10:04:10 Finished: FAILURE
```

**Comment by autopulated** [comment]
looks like a bug in the caching, you can probably workaround it by removing ~/.yotta/cache

**Comment by mazimkhan** [comment]
All Jenkins workspaces are created from scratch. there should not be a ~/.yotta/cache file. Are you suggesting it happens due to stale ~/.yotta/cache file.

**Comment by autopulated** [comment]
The point where the error occurs is where a downloaded module is being put into the cache, and the error implies the same file was already present (which is a case that should be handled), hence my suggestion, but maybe the file was put there by a parallel build, for example.

**Comment by mazimkhan** [comment]
Oh I think I overlooked the path. So the .yotta dir is in home directory. In automated builds I don't want to delete this file from home. As I don't know what other builds would be doing with it. Its a shared file created by yotta and yotta should be able to manage its states and contingencies.

Can we have a no cache option in yotta or we can specify the cache in the workspace that is local to the build. Perhaps later would defy the purpose of it.

**Comment by autopulated** [comment]
You normally want the cache, especially if you are building similar things many times, because it will reduce duplicate downloads. Configure the cache size with `"maxCachedModules":<number>` in your `~/.yotta/config.json` file: you could set this to zero to temporarily disable it.

**Comment by mazimkhan** [comment]
Yes the feature is good. But it does not support parallel instances of yotta. Do you think you would be fixing this issue near future. By that time we can disable the cache size.

**Comment by autopulated** [comment]
The cache *should* support parallel instances of yotta, that's why this is a bug :)

**Comment by mazimkhan** [comment]
Had updated ~/.yotta/config.json with ```"maxCachedModules": 0```
```
{
  "keys": {
    "private": xyz,
    "public": xyz,
  },
  "uuid": xyz,
  "mbed": {
    "authtoken": xyz
  },
  "maxCachedModules": 0
}
```

Still I am getting the same exception:

http://e108747.cambridge.arm.com:8080/job/ble-examples-nightly-build/label=gcc,target=nrf51dk-gcc/1/console

**Comment by autopulated** [comment]
OK thanks for trying that, I will look at this soon.

**Comment by autopulated** [comment]
Should be fixed by #713.
